blockprod/src/detail/tests.rs

```diff
@@ -185,26 +426,128 @@ async fn stop_job_non_existent_job(#[case] seed: Seed) {
     .expect("Error initializing blockprod");
 
     let (_other_job_key, _other_job_cancel_receiver) = block_production
-        .job_manager
+        .job_manager_handle
         .add_job(None, Id::new(H256::random_using(&mut rng)) as Id<GenBlock>)
         .await
         .unwrap();
 
-    let stop_job_key = JobKey::new(None, Id::new(H256::random_using(&mut rng)) as Id<GenBlock>);
+    let (_stop_job_key, _stop_job_cancel_receiver) = block_production
+        .job_manager_handle
+        .add_job(None, Id::new(H256::random_using(&mut rng)) as Id<GenBlock>)
+        .await
+        .unwrap();
 
-    let job_stopped = block_production.stop_job(stop_job_key).await.unwrap();
-    assert!(!job_stopped, "Stopped a non-existent job");
+    let jobs_stopped = block_production.stop_all_jobs().await.unwrap();
+    assert_eq!(jobs_stopped, 2, "Incorrect number of jobs stopped");
 
-    let jobs_count = block_production.job_manager.get_job_count().await.unwrap();
-    assert_eq!(jobs_count, 1, "Jobs count is incorrect");
+    let jobs_count = block_production.job_manager_handle.get_job_count().await.unwrap();
+    assert_eq!(jobs_count, 0, "Jobs count is incorrect");
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn stop_all_jobs_error() {
+    let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test(None);
+
+    let mut block_production = BlockProduction::new(
+        chain_config,
+        chainstate.clone(),
+        mempool,
+        Default::default(),
+        prepare_thread_pool(1),
+    )
+    .expect("Error initializing blockprod");
+
+    let mut mock_job_manager = Box::<MockJobManager>::default();
+
+    mock_job_manager
+        .expect_stop_all_jobs()
+        .times(1)
+        .returning(|| Err(JobManagerError::FailedToStopJobs));
+
+    block_production.set_job_manager(mock_job_manager);
+
+    let result = block_production.stop_all_jobs().await;
+
+    assert!(matches!(
+        result,
+        Err(BlockProductionError::JobManagerError(_))
+    ));
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn stop_all_jobs_ok(#[case] seed: Seed) {
+    let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test(None);
+
+    let mut block_production = BlockProduction::new(
+        chain_config,
+        chainstate.clone(),
+        mempool,
+        Default::default(),
+        prepare_thread_pool(1),
+    )
+    .expect("Error initializing blockprod");
+
+    let mut mock_job_manager = Box::<MockJobManager>::default();
+    let return_value = make_seedable_rng(seed).gen();
+    let _expected_value = return_value;
+
+    mock_job_manager
+        .expect_stop_all_jobs()
+        .times(1)
+        .returning(move || Ok(return_value));
+
+    block_production.set_job_manager(mock_job_manager);
+
+    let result = block_production.stop_all_jobs().await;
+
+    assert!(matches!(result, Ok(_expected_value)));
```

Comment:
matches! is redundant

Reply:
If I remove `matches!` here, `assert_eq!` will fail to compile. I've changed it into a vanilla `match`.

Reply:
Would it be solved by adding the `Eq` impl to the error type, if it's not there already?

Reply:
I thought about doing this, but didn't want to pollute the code and binary just for a single test. It's doing what it needs to without the `matches!` now, but if `Eq` is warranted, then I'm happy to add it.

Reply:
Cool, no problem.